tempesta_fw/t/functional/tests/test_parser.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
+
+__author__ = 'Tempesta Technologies'
+__copyright__ = 'Copyright (C) 2016 Tempesta Technologies. (info@natsys-lab.com).'
+__license__ = 'GPL2'
+
+import sys
+from os.path import dirname, realpath, sep
+sys.path.append((dirname(realpath(__file__)) + sep + "helpers"))
+
+import socket
+import tfw
+import conf
+import be
+import tfwparser
+import datetime
+import os
+import requests
+
+class Test:
+	def __init__(self):
+		self.res = ""
+		self.cfg = conf.Config('etc/tempesta_fw.conf')
+		self.cfg.add_option('cache', '0')
+		self.cfg.add_option('listen', '8081')
+		self.cfg.add_option('server', '127.0.0.1:8080')
+# Send request twice and check responses.
+	def run_with_cache(self):
+		self.res = "cache - two queries and compare bodies:\n"
+		self.cfg.del_option('cache')
+		self.cfg.add_option('cache', '1')
+		self.cfg.add_option('cache_fulfill', '* *')
+		self.resp = b'HTTP/1.0' + b' 200 - OK\r\n'
+		date = datetime.datetime.utcnow().strftime("%a, %d %b %Y" +\
+							   " %H:%M:%S GMT")
+		self.resp += b"Date: " + date + b"\r\n" 
+		self.resp += b"Server: be python\r\n\r\n"
+		self.resp += b'\r\n<html>content</html>\r\n\r\n'
+		self.run_test(2)
+		 
+# A response without the Content-Length header an without a body. 
+# For now (11.11.2016) tempesta's return status - 404.
```

Comment:
Ok, this is test for #619. But the comment is very strange and it seems the subject for investigation. It seems you have some misconfiguration - Tempesta must just return response without body in this test. Please learn the reason for 404 response and fix the tests.